core/js/typespecs/prompt.ts

```diff
@@ -95,6 +95,30 @@ export const responseFormatSchema = z.union([
   z.object({ type: z.literal("text") }).openapi({ title: "text" }),
 ]);
 
+export const responsesAPIJsonSchemaSchema = z.object({
+  type: z.literal("json_schema"),
+  name: z.string(),
+  description: z.string().optional(),
+  schema: z
+    .union([
+      z.record(customTypes.unknown).openapi({ title: "object" }),
+      z.string().openapi({ title: "string" }),
+    ])
+    .optional(),
+  strict: z.boolean().nullish(),
+});
+export type ResponsesAPIJsonSchema = z.infer<
+  typeof responsesAPIJsonSchemaSchema
+>;
+
+export const responsesAPIFormatSchema = z.union([
```

Comment:
when do you decide to add this type to the core typespecs vs. creating a spec where it is used?

Reply:
I have a feeling this is going to get reused in more places. But I wanted it to be next to the other response format definition.